test/blackbox-tests/test-cases/actions/actions-stdxxx-on-success.t

```diff
@@ -0,0 +1,221 @@
+Test for --actions-stdxxx-on-success
+====================================
+
+  $ export BUILD_PATH_PREFIX_MAP="sh=$(which sh):$BUILD_PATH_PREFIX_MAP"
+
+  $ echo '(lang dune 3.0)' > dune-project
+
+  $ cat > dune <<EOF
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Hello, world!'")))
+  > 
+  > (rule
+  >  (alias default)
+  >  (action (system "echo 'Something went wrong!' >&2")))
+  > EOF
+
+By default, stdout and stderr are always printed:
+
+  $ dune build
+            sh alias default
+  Hello, world!
+            sh alias default
+  Something went wrong!
+
+swallow tests
+-------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=swallow --actions-stderr-on-success=swallow
+
+must-be-be-empty tests
+----------------------
+
+  $ dune clean
+  $ dune build --actions-stdout-on-success=must-be-empty
+            sh alias default
+  Something went wrong!
+  File "dune", line 1, characters 0-65:
+  1 | (rule
+  2 |  (alias default)
+  3 |  (action (system "echo 'Hello, world!'")))
+            sh alias default (exit 0)
+  (cd _build/default && sh -c 'echo '\''Hello, world!'\''')
+  Hello, world!
+  [1]
```

Comment:
It may be unclear here why the build failed: could we give a better error message, explaining that the reason for failure is the `must-be-empty` setting?

Reply:
Same for the `stderr` test below.

Reply:
done

Reply:
The error messages still don't look ideal. Dune points to the action in question but doesn't explain what's wrong with it. It would be nice to print error messages like:

*  "The action's standard output must be empty but it is not" (for stdout)

*  "The action's standard error output must be empty but it is not" (for stderr)

Perhaps, we could also give a hint, like "Did you run Dune with --action-stdout-on-success=must-be-empty by mistake?"